coldfront/core/utils/management/commands/audit_data.py

```diff
@@ -0,0 +1,229 @@
+                    ' has a start date of {} that is different than its allocation period\'s.' \
+                        .format(id, project, start_date)))
+                if project_status == 'Active' and end_date != FCA_PCA_ALLOCATION_PERIOD.end_date:
+                    self.stdout.write(self.style.ERROR('Allocation {} for project {}'
+                    ' is active and has an end date of {} that is different than its allocation period\'s.' \
+                        .format(id, project, end_date)))
+                
+            elif project.startswith(ICA_PREFIX):
+                if project_status == 'Inactive' and end_date is not None:
+                    self.stdout.write(self.style.ERROR('Allocation {} for project {}'
+                    ' has an end date while being an inactive ICA.' \
+                        .format(id, project)))
+                if project_status == 'Active' and not any(end_date != ica.end_date for ica in ICA_ALLOCATION_PERIODS):
+                    self.stdout.write(self.style.ERROR('Allocation {} for project {}'
+                    ' is an active ICA that has an end date that is different from all ICA allocation periods.' \
+                        .format(id, project)))
+            elif project.startswith(RECHARGE_CONDO_PREFIX):
+                if end_date is not None:
+                    self.stdout.write(self.style.ERROR('Allocation {} for project {}'
+                    ' is a Recharge or Condo allocation with an end date.' \
+                        .format(id, project)))
+    
+    def handle_project_inactive(self):
+        '''
+        Assert that inactive Projects have "Expired" allocations and zero SUs.
+        '''
+        inactive_projects = Project.objects.prefetch_related('allocation_set__status').filter(status__name='Inactive') 
+        
+        for project in inactive_projects:
+            for allocation in project.allocation_set.select_related('status').prefetch_related('allocationattribute_set'):
+                if allocation.status.name != 'Expired':
+                    self.stdout.write(self.style.ERROR('Project {} ({})'
+                    ' is inactive and has unexpired allocations.' \
+                        .format(project.id, project.name)))
+                try:
+                    allocation_attribute = allocation.allocationattribute_set.get(allocation_attribute_type__name='Service Units')
+                    if allocation_attribute.value == 0:
+                        self.stdout.write(self.style.ERROR('Project {} ({})'
+                    ' is inactive and has non-zero SUs.' \
+                            .format(project.id, project.name)))
+                except AllocationAttribute.DoesNotExist as e:
+                    pass
+
+    def handle_project_pi(self):
+        '''
+        Assert that all projects have at least one PI.
+        '''
+        projects = Project.objects.all()
+        for project in projects:
+            if project.pis().count() == 0:
```

Comment:
Whenever you do queryset.count() == 0, you can also use queryset.exists(). It'll return true if the count > 0 and false otherwise